src/ModalPreviewLayout/test/ModalPreviewLayout.spec.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { createRendererWithUniDriver, cleanup } from '../../../test/utils/unit';
+
+import ModalPreviewLayout from '../ModalPreviewLayout';
+import { modalPreviewLayoutPrivateDriverFactory } from './ModalPreviewLayout.private.uni.driver';
+
+const requiredProps = {
+  children: <div />,
+  onClose: () => {},
+};
+
+describe('ModalPreviewLayout', () => {
+  const render = createRendererWithUniDriver(
+    modalPreviewLayoutPrivateDriverFactory,
+  );
+
+  afterEach(cleanup);
+
+  it('should render', async () => {
+    const { driver } = render(<ModalPreviewLayout {...requiredProps} />);
+
+    expect(await driver.exists()).toBeTruthy();
+  });
+
+  it('should call onClose when clicking outside by default', async () => {
+    const props = {
+      ...requiredProps,
+      onClose: jest.fn(),
+    };
+    const { driver } = render(<ModalPreviewLayout {...props} />);
+
+    await driver.clickOverlay();
+
+    expect(props.onClose).toHaveBeenCalled();
+  });
+
+  it('should not call onClose when shouldCloseOnOverlayClick set to false', async () => {
+    const props = {
+      ...requiredProps,
+      shouldCloseOnOverlayClick: false,
+      onClose: jest.fn(),
+    };
+    const { driver } = render(<ModalPreviewLayout {...props} />);
+
+    await driver.clickOverlay();
+
+    expect(props.onClose).not.toHaveBeenCalled();
+  });
+
+  it('should call onClose when clicking on the close button', async () => {
+    const props = {
+      ...requiredProps,
+      onClose: jest.fn(),
+    };
+    const { driver } = render(<ModalPreviewLayout {...props} />);
+
+    await driver.clickClose();
+
+    expect(props.onClose).toHaveBeenCalled();
+  });
+
+  it('should render the preview content', async () => {
+    const contentDataHook = 'preview-content';
+    const props = {
+      ...requiredProps,
+      children: <div data-hook={`${contentDataHook}`} />,
+    };
+    const { driver } = render(<ModalPreviewLayout {...props} />);
+
+    expect(await driver.contentByDataHook(contentDataHook)).not.toBeNull();
```

Comment:
I think that in order to make this test more reliable, we should verify the content instead of ensuring it's not null.

For example:
```javascript
 const props = {
      ...requiredProps,
      children: <div data-hook={`${contentDataHook}`}>Some content</div>,
    };
```

And then:

```suggestion
    expect(await driver.contentByDataHook(contentDataHook).text()).toBe('Some content');
```

Reply:
I copied from the messagebox specs, so it's a trade-off with consistency

Reply:
We've a lot of practices in the project and apparently will not achieve absolute consistency in the near future...

In this case I'd go with the better solution

Reply:
sure, np